docs/docs/adding-a-shopping-cart-with-snipcart.md

```diff
@@ -0,0 +1,197 @@
+
+Importantly, `data-item-url` denotes the URL of the webpage displaying the product(s). Snipcart needs to crawl this page to validate the order. The web crawler looks for the HTML element with the `snipcart-add-item` CSS class as well as the `data-item-id` and checks what it finds there against whatever is in the cart.
+
+> Note that, while you're testing, a `data-item-url` value of `"/"` is fine. For the checkout flow to work, you will eventually need to replace this with the actual URL at which you've published your catalog or product page.
+
+To learn more about defining products, see the [Snipcart documentation](https://docs.snipcart.com/v3/setup/products).
+
+### Adding product variants
+
+Snipcart refers to variations like size and color as "product options" or "custom fields". You can add these custom fields to your products to allow customers to refine their orders.
+
+Building on the stacking ring example, suppose that you wanted to give your customer a choice between available sizes. You would do this by adding a custom field with a name and options.
+
+```jsx
+<section>
+  <h2>Silver Stacking Ring</h2>
+  <p>$19.99</p>
+  <p>Wear one or seventeen! These rings are fun to mix and match.</p>
+  <button
+    class="snipcart-add-item"
+    data-item-id="silver-stacking-ring"
+    data-item-price="19.99"
+    data-item-url="/"
+    data-item-name="Silver Stacking Ring"
+    <!--highlight-start-->
+    data-item-custom1-name="Size"
+    data-item-custom1-options="6|6.5|7|7.5|8|8.5|9"
+    <!-- highlight-end -->
+    >
+    Add to cart
+  </button>
+</section>
+```
+
+You can add multiple custom fields by incrementing the index of the `data-item-custom` attribute. Perhaps you want customers to have the ability to mark each item in their cart as a gift.
+
+```jsx
+<section>
+  <h2>Silver Stacking Ring</h2>
+  <p>$19.99</p>
+  <p>Wear one or seventeen! These rings are fun to mix and match.</p>
+  <button
+    class="snipcart-add-item"
+    data-item-id="silver-stacking-ring"
+    data-item-price="19.99"
+    data-item-url="/"
+    data-item-name="Silver Stacking Ring"
+    data-item-custom1-name="Size"
+    data-item-custom1-options="6|6.5|7|7.5|8|8.5|9"
+    <!--highlight-start-->
+    data-item-custom2-name="This is a gift" data-item-custom2-type="checkbox"
+   >
+    <!-- highlight-end -->
+    Add to cart
+  </button>
+</section>
+```
+
+### Selling digital products
+
+Snipcart enables the sale of digital goods. To sell a file, you'll need to upload it to your Snipcart dashboard and then add the resulting guid as the value of the `data-item-file-guid` attribute to your product's markup. You can specify a file access expiry in days and a maximum number of downloads per order from the dashboard.
```

Comment:
Not sure I understand "file" in this context?

Reply:
As in a file you would like people to pay to download. I felt like "digital good" was a bit vague but that's how they describe it in the Snipcart docs. Open to suggestions.

Reply:
Ahh, ok, so the heading context was totally lost on me. And I just assumed digital goods meant buying stuff via the web rather than the product itself being downloadable. Perhaps that's what needs to be clarified? So `Snipcart enables the sale of digital goods that consumers download directly to their computer.`? It's redundant but helps hammer home the shift from online shopping to online goods.